executor.py

```diff
@@ -1,18 +1,49 @@
 from jina import DocumentArray, Executor, requests
 from InstructorEmbedding import INSTRUCTOR
+from typing import Dict
+import torch
 
 class InstructorEmbeddingExecutor(Executor):
     """InstructorEmbeddingExecutor embeds texts into 768-dim vectors using instructor embedding"""
-    def __init__(self, **kwargs):
+    def __init__(
+            self,
+            model_name: str = 'hkunlp/instructor-large',
+            batch_size: int = 32,
+            output_value: str = 'sentence_embedding',
+            convert_to_numpy: bool = True,
+            convert_to_tensor: bool = False,
+            normalize_embeddings: bool = False,
+            device: str = 'cpu',
+            **kwargs):
         super().__init__(**kwargs)
-        self.model = INSTRUCTOR('hkunlp/instructor-large')
+        self.batch_size = batch_size
+        self.output_value = output_value,
+        self.convert_to_numpy = convert_to_numpy,
+        self.convert_to_tensor = convert_to_tensor,
+        self.normalize_embeddings = normalize_embeddings,
+        self.model = INSTRUCTOR(model_name_or_path=model_name, device=device)
+
     @requests
     def foo(self, docs: DocumentArray, **kwargs):
         pass
 
     @requests(on='/')
-    def generate_embedding(self, docs: DocumentArray, **kwargs):
-        for i in docs:
-            for j in i.matches:
-                j.embedding = self.model.encode([[i.text, j.text]], convert_to_tensor=True)[0]
-        return docs
\ No newline at end of file
+    def encode(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        for doc in docs:
+            # doc.text : instruction
+            # doc.matches : sentences
+            document_batches_generator = DocumentArray(
```

Comment:
# input of instruction embedder

`(instruction, text)`

# docarray format 

`doc = Document(text='blabalal', tags={'instruction': 'encode for science....'})`

# parse input docarray from request

```
@request
def encode(self, docs: DocumentArray, params: dict, **kwargs):
    for mini_batch in docs.batch(batch_size=parameters.get('batch_size', 16)):
        batch_input = [(d.tags['instruction'], d.text) for d in mini_batch if d.tags.get('instruction') and d.text]
        with torch.inference_model():
             mini_batch.embeddings = self.model.encode(batch_input, .....)
```

Reply:
# instruction is pass by request params
```
instruction = doc.tags.get('instruction', None) or prameters.get('instruction', '')
```